redis/src/auth_management.rs

```diff
@@ -125,3 +125,74 @@ pub mod credentials_management_utils {
         Err("OID claim not found".to_string())
     }
 }
+
+#[cfg(all(feature = "token-based-authentication", test))]
+mod auth_management_tests {
```

Comment:
since the tests are here, then things that shouldn't be `pub` can be just `pub(crate)` or even private in the file above. please ensure everything is minimally public.